src/System.Runtime.Serialization.Xml/tests/Performance/PerformanceTestsCommon.cs

```diff
@@ -0,0 +1,207 @@
+            {
+                for (int i = 0; i < childListSize; ++i)
+                {
+                    obj.SimpleTypeList.Add(CreateObject(height - 1, index, i, collectionSize, childListSize));
+                }
+            }
+            return obj;
+        }
+
+        public static byte[] CreateByteArray(int size)
+        {
+            byte[] obj = new byte[size];
+            for (int i = 0; i < obj.Length; ++i)
+            {
+                obj[i] = (byte)(i % 256);
+            }
+            return obj;
+        }
+
+        public static Dictionary<int, string> CreateDictionaryOfIntString(int count)
+        {
+            Dictionary<int, string> dictOfIntString = new Dictionary<int, string>();
+            for (int i = 0; i < count; ++i)
+            {
+                dictOfIntString[i] = i.ToString();
+            }
+
+            return dictOfIntString;
+        }
+
+        public static List<int> CreateListOfInt(int count)
+        {
+            return Enumerable.Range(0, count).ToList();
+        }
+
+        public static object GetSerializationObject(TestType testType, int testSize)
+        {
+            Object obj = null;
+            switch (testType)
+            {
+                case TestType.ByteArray:
+                    obj = CreateByteArray(testSize);
+                    break;
+                case TestType.Dictionary:
+                    obj = CreateDictionaryOfIntString(testSize);
+                    break;
+                case TestType.ISerializable:
+                    obj = new ClassImplementingIXmlSerialiable() { StringValue = "Hello world" };
+                    break;
+                case TestType.ListOfInt:
+                    obj = CreateListOfInt(testSize);
+                    break;
+                case TestType.SimpleType:
+                    obj = CreateObject(testSize, 0, -1, 7, 2);
+                    break;
+                case TestType.String:
+                    obj = new string('k', testSize);
+                    break;
+                case TestType.XmlElement:
+                    XmlDocument xmlDoc = new XmlDocument();
+                    xmlDoc.LoadXml(@"<html></html>");
+                    XmlElement xmlElement = xmlDoc.CreateElement("Element");
+                    xmlElement.InnerText = "Element innertext";
+                    obj = xmlElement;
+                    break;
+            }
+            return obj;
+        }
+
+        #endregion
+
+        #region Methods to run serialization performance tests
+
+        public static void RunSerializationPerformanceTest(int iterations, SerializerType serializerType, TestType testType, int testSize, SerializerFactory serializerFactory)
+        {
+            var obj = GetSerializationObject(testType, testSize);
+
+            var serializer = serializerFactory.GetSerializer(serializerType);
+            serializer.Init(obj);
+
+            using (var stream = new MemoryStream())
```

Comment:
MemoryStream starts with a size of 0, then allocates 256 bytes on the first write. It then doubles in size and needs to copy the contents each time you exceed the capacity. As you are continuously writing to the same stream without resetting it with each iteration, you are causing the stream to hold all results of all tests. The ideal thing would be to know the maximum size stream you would need and preallocate a byte array before the test starts and pass to the MemoryStream constructor. This way MemoryStream will throw an exception if you try to exceed that and you guarantee there's no allocation coming from MemoryStream. An alternative solution would be to do a priming serialization after the MemoryStream has been allocated so that MemoryStream has grown sufficiently to avoid allocations before measurements begin. You could still end up with GC from any allocations potentially happening during the test, but it would be a single GC. Regardless which option you use, you should reset the stream before each serialization with `stream.Position = 0;`.